### PR TITLE
Rewrite documentation

### PR DESCRIPTION
**Pull request by jason-fox**
- Move excessive README into the documentation so it can be searched
   - Architecture
  - API documentation  - also restructured, added information on data models
  - Advanced Topics - add metadata section
- Add new Background
- Add new Getting started
- General Tidy up of texts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,1172 +29,72 @@ check the FIWARE Catalogue entry for the
...
-Returns:
-
--   200 OK if successful, with no payload.
--   400 MISSING_HEADERS if any of the mandatory headers is not present.
--   400 WRONG_SYNTAX if the body doesn't comply with the schema.
--   500 SERVER ERROR if there was any error not contemplated above.
-
-##### GET /iot/services
+The main concept of the **IoT Agent node library** is to provide a common framework for provisioning IoT devices,
+allowing each individual IoT Agent to access standardized mapping data for devices and to offer a series common utility
```
Comment (AlvaroVega) [inline README.md]: standardized -> standarized

File: doc/advanced-topics.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
+## Advanced Topics
```
Comment (fgalan) [inline doc/advanced-topics.md]: I understand this is a new document with content previously on README.md about advance topics.

I see the new document is linked from mkdocs.yml, but it should be also linked from README.md and I haven't found that link (maybe I missed?).

File: doc/architecture.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
+## Architecture
```
Comment (fgalan) [inline doc/architecture.md]: Similarly to advanced_topics.md, this document should be linked from the top README.md.

File: doc/development.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
+## Development documentation
```
Comment (fgalan) [inline doc/development.md]: Similarlo to advanced_topics.md and architecture.md this document should be linked from the toplevel README.md.

Comment (fgalan) [reply doc/development.md]: ...toplevel README.md or maybe other

File: doc/usermanual.md [hunk 5]
```diff
@@ -1,218 +1,238 @@
...
-
+# Library Functions
+
+-   [Usage](#usage)
+    -   [Stats Registry](#stats-registry)
+    -   [Alarm module](#alarm-module)
+    -   [Transactions](#transactions)
+    -   [Library overview](#library-overview)
+    -   [Function reference](#function-reference)
+-   [Development documentation](#development-documentation)
```
Comment (fgalan) [inline doc/usermanual.md]: Not sure but it seems this piece of docs has been moved to a separate document (usermanual.md), so the sub-ToC here wouldn't make too much sense...

File: doc/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
+## Getting Started
```
Comment (fgalan) [inline doc/getting-started.md]: This document is entererly new (I mean, not moved from some other in this PR)? Fine with it, just to know...

In addition, it should be linked from README.md.

File: package.json [hunk 7]
```diff
@@ -63,6 +65,14 @@
     "nock": "10.0.6",
     "should": "13.2.3",
     "timekeeper": "2.2.0",
-    "watch": "~1.0.2"
+    "watch": "~1.0.2",
+    "remark-cli": "^6.0.1",
```
Comment (fgalan) [inline package.json]: Please use ~ instead of ^

File: package.json [hunk 8]
```diff
@@ -33,6 +33,8 @@
     "test": "mocha --recursive 'test/**/*.js' --reporter spec --timeout 3000 --ui bdd --exit",
     "test:watch": "npm run test -- -w ./lib",
     "lint": "jshint lib/ --config .jshintrc && jshint test/ --config test/.jshintrc",
+    "lint:md": "remark  -f '*.md' 'doc/*.md'",
+    "lint:text": "textlint '*.md' 'doc/*.md'",
```
Comment (fgalan) [inline package.json]: Should be mentioned in the document about npm tasks (development.md?), isn't it?

File: doc/advanced-topics.md (hunk 2 above)
Comment (jason-fox) [reply doc/advanced-topics.md]: Readme updated. Fix c45ba05

File: doc/architecture.md (hunk 3 above)
Comment (jason-fox) [reply doc/architecture.md]: Readme updated. Fix c45ba05

File: doc/development.md (hunk 4 above)
Comment (jason-fox) [reply doc/development.md]: Readme updated. Fix c45ba05

File: doc/getting-started.md (hunk 6 above)
Comment (jason-fox) [reply doc/getting-started.md]: Readme updated. Fix c45ba05

File: package.json (hunk 7 above)
Comment (jason-fox) [reply package.json]: Fix c45ba05

File: package.json (hunk 8 above)
Comment (jason-fox) [reply package.json]: development.md updated. Fix c45ba05

File: doc/usermanual.md (hunk 5 above)
Comment (jason-fox) [reply doc/usermanual.md]: Fix 9844afe

**Comment by jason-fox** [comment]
> LGTM. Thanks!
> 
> Should I merge? Or is there any pending adjustment @jason-fox ?

please merge and check that readthedocs builds the new documentation

**Comment by fgalan** [comment]
I have had a look to https://iotagent-node-lib.readthedocs.io and seems to be ok. Addional checks are welcome.

**Pull request merged**